### PR TITLE
Implement MLP and gridsearch

### PR DESCRIPTION
**Pull request by yli5**
1. Implement MLP with following tricks:
    (1) Mini-batch Learning
    (2) Batch Normalization
    (3) Early Stopping
    (4) Weight/Bias Initialization
    (5) TensorBoard for Visualization

2. Implement GridSearch to optimize Neural Network archtectures.

### CODE REVIEW

**Review by jkim- — APPROVED** [review]
Looks good
